### PR TITLE
[IMP] make the code more inheritable

### PR DESCRIPTION
**Pull request by sebastienbeau**
Code refactor to make the code more inheritable

### CODE REVIEW

**Review by ivantodorovich — APPROVED** [review]

File: shopinvader/services/cart.py [hunk 1]
```diff
@@ -317,13 +320,13 @@ def _add_item(self, cart, params):
...
-                self.env["sale.order.line"].create(vals)
+                    vals["name"] = self._get_sale_order_line_name(vals)
+                item = self.env["sale.order.line"].create(vals)
             cart.recompute()
+        return item
 
-    def _get_sale_order_line_name(self, product_id):
-        product = self.env["product.product"].browse(product_id)
+    def _get_sale_order_line_name(self, vals):
+        product = self.env["product.product"].browse(vals["product_id"])
```
Comment (simahawk) [inline shopinvader/services/cart.py:332]: you can pass the record here to not browse again

**Review by simahawk — APPROVED** [review]

**Review by hparfr — APPROVED** [review]
Rebased

**Pull request merged**